tests/lit-tests/loop_unroll_varying.ispc

```diff
@@ -0,0 +1,158 @@
+// Test to check '#pragma unroll' functionality for different loop statements
+//   with varying type induction variables.
+
+// RUN: %{ispc} %s --target=host --nowrap -O0 --emit-llvm-text --no-discard-value-names --nostdlib -o - | FileCheck %s -check-prefixes=CHECK_ALL,CHECKO0
+// RUN: %{ispc} %s --target=host --nowrap -O2 --emit-llvm-text --no-discard-value-names --nostdlib -o - | FileCheck %s -check-prefixes=CHECK_ALL,CHECKO2
+
+// XFAIL: *
+
+
+#define NUM 1024
+
+
+// CHECK_ALL-LABEL: define void @foo_for___
+
+// CHECKO0: call void @goo_for___
+// CHECKO0: br label %{{[a-zA-Z_][a-zA-Z0-9_]*}}, !llvm.loop
+
+// CHECKO2: call void @goo_for___
+// CHECKO2: call void @goo_for___
+// CHECKO2: call void @goo_for___
```

Comment:
If we'd like to check that we have exactly 3 calls, we should check that it's not match after that. Also, -COUNT-N suffix is more compact form of writing N same statements, i.e.:
```
// CHECKO2-COUNT-3: call void @goo_for___
// CHECKO2-NOT: call void @goo_for___
```